src/infra/docs/cdn.md

```diff
@@ -0,0 +1,88 @@
+# Content Delivery Networks
+
+Users of the Rust programming language interact with the infrastructure of the
+project in various different ways. They access the project's website and
+documentation, query the crates index, and download Rust releases and crates.
+These resources are hosted by the Rust project and served through a [Content
+Delivery Network] (CDN).
+
+This document outlines why we use CDNs, for what, and how we have set them up.
+
+## Objectives
+
+We have three goals for our use of CDNs in our infrastructure:
+
+1. Reduce costs of outbound traffic through cheaper pricing and caching
+2. Reduce load on origin servers to save compute resources
+3. Provide a way to rewrite legacy URLs for some resources
```

Comment:
in general: reduce cost / reduce load sounds is perhaps not enough, should we include _improve site speed for users_? 

also, coming from docs.rs: 
- serve content from a location more near to the users ([saves ~100ms](https://zhiguang.me/2016/05/10/latency-between-aws-global-regions/) ) 
- a way to potentially rewrite content ( [client-side CSP token generation could be an example](https://github.com/syphar/docs-rs-fastly-csp/blob/main/src/main.rs) ) (limited with cloudfront)

With fastly we could be more flexible with also caching more API responses , leveraging their fast invalidation times. Though I also get the reliability aspect of going multi-CDN.

Reply:
Thanks, @syphar, that's useful context!

Even though it's not a great answer, right now we are really looking at managing our costs. Since the pull request is intended to describe the current state, I'm hesitant to add anything to these three objectives since it doesn't feel genuine.

The next step for us is to implement quick wins to reduce our AWS bill. Which is why we're looking at `static.crates.io` and `static.rust-lang.org` as the most used distributions in particular. More details coming in another PR on top of this one, but don't expect any new features just yet.

I think that once we've taken care of that, we can have a broader discussion about how we can leverage the CDNs for more than just cheaper outbound traffic. Our naive approach was to keep things simple and just balance traffic, but it seems that both crates.io and docs.rs are interested in some of Fastly's features.

Let's have a separate discussion how we can support that after we've made some progress on the static distributions.

Reply:
thanks for the context here!